packages/@react-spectrum/overlays/src/Underlay.tsx

```diff
@@ -22,6 +22,7 @@ export function Underlay({isOpen, ...props}: UnderlayProps) {
   return (
     <div 
       {...filterDOMProps(props)}
+      tabIndex={-1}
```

Comment:
Alternative was to call `useOverlay` in https://github.com/adobe-private/react-spectrum-v3/blob/master/packages/@react-spectrum/overlays/src/Modal.tsx#L33, then pass the overlay's `onkeydown` to `Underlay` as well. Went with this solution since it felt simpler and I found mention of the above approach in some older w3 docs.

Lemme know if there are any accessibility concerns about the above approach and I can change it to the above approach

Reply:
I've seen mention of this as well, seems fine to me. @majornista ?

Reply:
Isn't this the opposite? Adding a `tabIndex` makes it focusable...

Reply:
Ah yeah that is true but it also removes it from being focusable via click or keyboard command right? The problem previously was that the body of the page was getting focused when clicking on the Underlay, making it so the escape key doesn't close the modal since it lost focus

Reply:
`tabIndex="-1"` makes it focusable via the mouse, but not tabbable. Can you look into why FocusScope is not handling the focus event of the body and moving focus back into the dialog like it should?

Reply:
Derp, ok did some brushing up on focus behavior and did some more digging. Looks like clicking on the body doesn't trigger the FocusScope's `onFocus` which is responsible for moving the focus back into the dialog. According to https://allyjs.io/data-tables/focusable.html (looks a bit out of date tho), `body` is technically not focusable but gets focus if no other elements have focus and it won't emit a focus event when it becomes `document.activeElement`. Adding a `tabIndex=-1` to the body makes it focusable via click hence making the `onFocus` trigger properly, returning focus back to the dialog. This is also why adding the `tabIndex=-1` to Underlay works as well.

Not 100% sure about the `body` focusable bit, seeing Microsoft docs say something different, hrmm

Reply:
Ah, maybe we need to handle blur as well as focus. The [focusout](https://developer.mozilla.org/en-US/docs/Web/API/Element/focusout_event) should do what we need. Check `event.relatedTarget`, and if its outside the current scope, move focus back into the scope.

Reply:
hrm, ran into some issues. What if the FocusScope is wrapping some input fields and the user clicks onto an arbitrary outside element? Doing the above would make the FocusScope unescapable. I thought changing the check to be if `event.relatedTarget = null` (i.e. a non-focusable element) but that didn't feel great either

Reply:
Ah actually this issue happens already due to `onFocus`, is FocusScope meant to be used exclusively in modals/popovers/elements that close when any interaction outside of it's scope happens?

Reply:
Yeah, that's what FocusScope is supposed to do (with the `contain` prop). It traps focus within the scope so it cannot be escaped.